### ISSUE TITLE
Android: setting the swiper index on render

### ISSUE THREAD

**Issue by tasharofi** [leecade/react-native-swiper#463]
I am having a swiper on my app for which the index is set dynamically on the scene render/load. All working fine on iOS but on android, even though the swiper bullet is set to the right one (the dots below the slides) but the actual slide is always set to the first one. 
I did some search on the already raised issues regarding this but none of them helped me to resolve my problem as I think my case is a bit different as the index needs to be set on scene render (I am not updating it after the render. I need it to be set on scene load).

can anyone please help me out with this issue?

Thanks,

**Comment by tasharofi** [comment]
Anyone help please?

**Comment by jarretmoses** [comment]
@tasharofi if you have a reference to the swiper (and haven't already tried this) you could try `swiper.scrollBy(PUT_OFFSET_HERE, false);` this should set your swiper to a certain position where the false means no animation. While this may not be the best solution it could be worth a shot.

**Comment by tasharofi** [comment]
Yes even though it doesn't produce the best user experience but I ended up using scrollBy in componentDidMount. 

Just for the reference and save some headache for others, I had to call this action with some delay (1000 ms) using setTimeout.

@jarretmoses  Thanks for the response anyway.

**Comment by shafy** [comment]
You can also use the `index` prop of the Swiper component to set the Slide you want upon rendering. It works for me also on Android.

**Comment by tasharofi** [comment]
@shafy the index prop would not in this scenario on Android. Doesnt pick up the index in time.

**Comment by shafy** [comment]
@tasharofi Hmm weird, it works for me. What version are you on?
